backend/app/services/scheduled_inquiries.py

```diff
@@ -1,32 +1,103 @@
+
 
 def create(*, session: Session, inquiry_id: int) -> ScheduledInquiryPublic:
     highest_rank = session.exec(
         select(ScheduledInquiry.rank).order_by(desc(ScheduledInquiry.rank))
     ).first()
-
     rank = highest_rank + 1 if highest_rank else 1
-
     scheduled_inquiry = ScheduledInquiryBase(inquiry_id=inquiry_id, rank=rank)
-
     db_inquiry = ScheduledInquiry.model_validate(scheduled_inquiry)
-
     session.add(db_inquiry)
     session.commit()
     session.refresh(db_inquiry)
-
     return ScheduledInquiryPublic.model_validate(db_inquiry)
 
 
+def enable_scheduled_inquiry(
+    *, session: Session, scheduled_inquiry_id: int
+) -> ScheduledInquiryPublic:
+    stmt = select(ScheduledInquiry).where(ScheduledInquiry.id == scheduled_inquiry_id)
+    target: ScheduledInquiry | None = session.exec(stmt).first()
+    if not target:
+        raise ValueError("Invalid scheduled_inquiry_id to enable")
+    highest_rank = session.exec(
+        select(ScheduledInquiry.rank).order_by(desc(ScheduledInquiry.rank))
+    ).first()
+    try:
+        target.sqlmodel_update({"rank": highest_rank + 1 if highest_rank else 1})
+        session.commit()
+    except Exception:
+        raise ValueError("Invalid scheduled inquiry update request")
+    session.refresh(target)
+    return ScheduledInquiryPublic.model_validate(target)
+
+
+def disable_scheduled_inquiry(
+    *, session: Session, scheduled_inquiry_id: int
+) -> ScheduledInquiryPublic:
+    stmt = select(ScheduledInquiry).where(ScheduledInquiry.id == scheduled_inquiry_id)
+    target: ScheduledInquiry | None = session.exec(stmt).first()
+    if not target:
+        raise ValueError("Invalid scheduled_inquiry_id to disable")
+    try:
+        target.sqlmodel_update({"rank": 0})
+        session.commit()
+    except Exception:
+        raise ValueError("Invalid scheduled inquiry update request")
+    session.refresh(target)
+    return ScheduledInquiryPublic.model_validate(target)
+
+
+def update_scheduled_inquiry(
+    *, session: Session, scheduled_inquiry_in: ScheduledInquiryUpdate
+) -> ScheduledInquiryPublic:
+    stmt = select(ScheduledInquiry).where(
+        ScheduledInquiry.id == scheduled_inquiry_in.id
+    )
+    target: ScheduledInquiry | None = session.exec(stmt).first()
+    if not target:
+        raise ValueError("Invalid inquiry.id")
+    move_rank = (
+        scheduled_inquiry_in.rank - target.rank
+        if scheduled_inquiry_in.rank > 0
+        and target.rank > 0
+        and abs(scheduled_inquiry_in.rank - target.rank) == 1
+        else 0
+    )
+    if move_rank != 0:
+        # update rank of the scheduled_inquiry currently at target rank
+        move_rank_stmt = (
+            update(ScheduledInquiry)
+            .where(ScheduledInquiry.rank == scheduled_inquiry_in.rank)  # type: ignore [arg-type]
+            .values(rank=ScheduledInquiry.rank - move_rank)
+        )
+        session.exec(move_rank_stmt)  # type: ignore [call-overload]
```

Comment:
wouldn't we also have to update rank of all the other inquiries? i.e. if you have 1,2,3,4,5, and move 5 to 3, then 3->4, 4->5 ?

Reply:
one way to get around this, btw, may be to use decimals for rank. So in this example, you start with 1,2,3,4,5, move 5->3 means 1,2,3, 3.5, 4. Another 5->3 means 1,2,3, 3.25, 3.5. Then we can do like a daily job to de-decimalize the stack.

Reply:
This update is only triggered when a rank is between greater than 1,  less than MAX and only moving a single position. This allows it to work by updating only the inquiry at the target rank and itself, essentially trading places. This was a first pass at implementing basic move rank behavior. Running the application, you can see it works for current UI. I had considered using float field as well, but rank already existed and I thought this was good initial implementation. The logic would definitely need adjusted for drag and drop UI.